tutorials/physics/using_character_body_2d.rst

```diff
@@ -497,91 +469,68 @@ Here's the code for the player body:
 
     extends CharacterBody2D
 
-    export (int) var run_speed = 100
-    export (int) var jump_speed = -400
-    export (int) var gravity = 1200
+    var speed = 300.0
+    var jump_speed = -400.0
 
-    var velocity = Vector2()
-    var jumping = false
+    # Get the gravity from the project settings so you can sync with rigid body nodes.
+    var gravity = ProjectSettings.get_setting("physics/2d/default_gravity")
 
-    func get_input():
-        velocity.x = 0
-        var right = Input.is_action_pressed('ui_right')
-        var left = Input.is_action_pressed('ui_left')
-        var jump = Input.is_action_just_pressed('ui_select')
-
-        if jump and is_on_floor():
-            jumping = true
-            velocity.y = jump_speed
-        if right:
-            velocity.x += run_speed
-        if left:
-            velocity.x -= run_speed
 
     func _physics_process(delta):
-        get_input()
+        # Add the gravity.
         velocity.y += gravity * delta
-        if jumping and is_on_floor():
-            jumping = false
-        velocity = move_and_slide(velocity, Vector2(0, -1))
+
+        # Handle Jump.
+        if Input.is_action_just_pressed("jump") and is_on_floor():
+            velocity.y = jump_speed
+
+        # Get the input direction.
+        var direction = Input.get_axis("ui_left", "ui_right")
+        velocity.x = direction * speed
+
+        move_and_slide()
 
  .. code-tab:: csharp
 
     using Godot;
 
-    public partial class KBExample : CharacterBody2D
+    public partial class CBExample : CharacterBody2D
     {
-        [Export] public int RunSpeed = 100;
-        [Export] public int JumpSpeed = -400;
-        [Export] public int Gravity = 1200;
+        public float Speed = 100.0f;
+        public float JumpSpeed = -400.0f;
 
-        Vector2 velocity = new Vector2();
-        bool jumping = false;
+        // Get the gravity from the project settings so you can sync with rigid body nodes.
+        public float Gravity = ProjectSettings.GetSetting("physics/2d/default_gravity");
```

Comment:
`ProjectSettings.GetSetting` returns a `Variant` so it needs to be converted to `float`, you can cast it or use the `AsSingle` method, we use the latter in the [`basic_movement.cs`](https://github.com/godotengine/godot/blob/cd3e03432af9a638e342b5b79f76ca4d72f2a1f4/modules/mono/editor/script_templates/CharacterBody2D/basic_movement.cs#L12) template.

```suggestion
        public float Gravity = ProjectSettings.GetSetting("physics/2d/default_gravity").AsSingle();
```